Sprint 08

Kolesnikov-Eugene opened the pull request:
Не уверен насчет правильности реализации градиента внизу загружаемого изображения.
Перепробовал кучу вариантов, в том числе через создание дополнительного UIView через сториборд (наследуемого от созданного класса GradientView, в котором была реализована логика добавления градиента к ячейке таблицы).
В итоге оставил вариант с реализацией в методе addGradient в контроллере. Если будут комментарии, или наводка, по какому пути двигаться, чтобы правильно реализовать градиент -  буду очень рад помощи!))

Kolesnikov-Eugene committed:
added auto height of tableView cell, dateLabel and likeButton linked with imagesCell

Kolesnikov-Eugene committed:
added gradient to the bottom of ImageListCell

Kolesnikov-Eugene committed:
renamed folder ImagesList

helecta commented on FunImages/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,104 @@
+        
+        cell.dateLabel.text = dateFormatter.string(from: Date())
+        
+        let likeIsOn = indexPath.row % 2 == 0
```
По заданию лайк должен быть проставлен у чётных ячеек и не проставлен у нечётных, у тебя получилось наоборот. Напомню, что indexPath.row начинается с 0, а не 1.
Это не сильно критично, если будет желание можешь исправить

helecta commented on FunImages/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,104 @@
+}
+// MARK: - TableView Delegate
+extension ImagesListViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```
Пустые методы лучше удалять, так кода становится меньше и его проще читать

helecta commented on FunImages/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,104 @@
+
+        let gradientlayer = CAGradientLayer()
+        gradientlayer.colors = [
+          UIColor(red: 0.102, green: 0.106, blue: 0.133, alpha: 0).cgColor,
```
Лучше вынести цвета в Assets, тогда можно будет использовать цвета так:
```
        let colorTop = UIColor(named: "gradientTop")?.cgColor ?? UIColor.clear.cgColor
        let colorBottom = UIColor(named: "gradientBottom")?.cgColor ?? UIColor.clear.cgColor
```

helecta commented on FunImages/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,104 @@
+        addGradient(for: cell)
+    }
+    
+    func addGradient(for cell: ImagesListCell) {
```
Лучше перенести этот метод в класс ячейки `ImagesListViewController` и дальше есть несколько вариантов:
- либо добавить свойство, которое показывает есть ли уже слой с градиентом или нет, если есть не добавлять
- либо создавать слой один раз при инициализации ячейки (не конфигурации)
- либо удалять старый слой в конфигурации. Перед этим его сохранять для ячейки. 
Можно удалять слой в специальном методе `prepareForReuse`, который вызывается перед началом переиспользования. 

Если возникнут вопросы, смело задавай их Наставнику!

Kolesnikov-Eugene committed:
1. changed the order of like_button_on pictures.
2. Colors of Gradient added to assets
3. addGradient func moved to ImageListClass, added property hasGradient to check if cell already has a gradient added
